ada_feeding_perception/ada_feeding_perception/segment_all_items.py

```diff
@@ -0,0 +1,1279 @@
+        # [top left x-value, top left y-value, bottom right x-value, bottom right y-value]
+        # and unnormalize the bounding box coordinate values
+        boxes_xyxy = {}
+        for phrase, boxes in boxes_cxcywh.items():
+            boxes_xyxy[phrase] = []
+            for box in boxes:
+                # Scale the box from percentage values to pixel values
+                box = np.multiply(box, np.array([width, height, width, height]))
+                center_x, center_y, w, h = box
+                # Get the bottom left and top right coordinates of the box
+                x0 = center_x - (w / 2)
+                y0 = center_y - (h / 2)
+                x1 = x0 + w
+                y1 = y0 + h
+                boxes_xyxy[phrase].append([x0, y0, x1, y1])
+
+        # Measure the elapsed time running GroundingDINO on the image prompt
+        inference_time = int(round((time.time() - inference_time) * 1000))
+
+        return boxes_xyxy
+
+    def load_image(self, image_array: npt.NDArray):
+        """
+        Load the image and apply transformations to it.
+
+        Parameters
+        ----------
+        image_array: The image to load.
+
+        Returns
+        -------
+        image_pil: The image in Image pillow format.
+        image: The image in tensor format.
+        """
+        # Convert image from BGR to RGB to convert CV2 image to image pillow
+        image_array = cv2.cvtColor(image_array, cv2.COLOR_BGR2RGB)
+
+        # Convert image to image pillow to apply transformation
+        image_pil = ImagePIL.fromarray(image_array, mode="RGB")
+        transform = T.Compose(
+            [
+                T.RandomResize([800], max_size=1333),
+                T.ToTensor(),
+                T.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+            ]
+        )
+        image, _ = transform(image_pil, None)  # 3, h, w
+
+        return image_pil, image
+
+    def generate_mask_msg(
+        self,
+        item_id: str,
+        object_id: str,
+        score: float,
+        mask: npt.NDArray[np.bool_],
+        image: npt.NDArray,
+        depth_img: npt.NDArray,
+        bbox: Tuple[int, int, int, int],
+    ) -> Optional[Mask]:
+        """
+        Convert a mask detected by EfficientSAM or SAM into a ROS Mask message.
+
+        Parameters
+        ----------
+        item_id: The item ID of the mask.
+        object_id: The object ID of the mask.
+        score: The confidence score outputed by the segmentation model for the mask.
+        mask: The pixel-wise mask detected.
+        image: The image the mask was detected on.
+        depth_img: The most recent depth image.
+        bbox: The bounding box from GroundingDINO.
+        """
+        # Calculate center of the bounding box and use it as the seed point for
+        # getting the connected component of the mask
+        center_x = (bbox[0] + bbox[2]) // 2
+        center_y = (bbox[1] + bbox[3]) // 2
```

Comment:
For this and other instances of `bbox`, does it have any properties like `xmin`, `xmax`, `ymin`, `ymax` to make this easily understandable?

Reply:
You coudl consider using a [namedtuple](https://docs.python.org/3/library/collections.html#collections.namedtuple) for this